tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -1,90 +1,329 @@
+	instance, err := updateCreatedInstance(createdInstanceID)
+	if err != nil {
+		tearDownInstance(t, "error getting ec2 instance object after creation", err)
+	} else {
+		createdInstance = instance
+	}
+	assert.Equal(t, ec2.InstanceStateNameRunning, *createdInstance.State.Name, "instance is not in running state")
+
+	assert.Equalf(t, imageID, *createdInstance.ImageId, "instance image ID: %s", *createdInstance.ImageId)
+
+	assert.Equalf(t, instanceType, *createdInstance.InstanceType, "instance type %s", *createdInstance.InstanceType)
+
+	assert.Equalf(t, sshKey, *createdInstance.KeyName, "instance ssh key name: %s", *createdInstance.KeyName)
+}
+
+// checkInstanceHasPublicSubnetAndIp asserts if the instance is associated with a public IP address and  subnet by
+// checking if the subnet routing table has internet gateway attached.
+func checkInstanceHasPublicSubnetAndIp(t *testing.T) {
+	assert.NotEmpty(t, createdInstance.PublicIpAddress, "instance does not have a public IP address")
+
+	routeTables, err := awsProvider.EC2.DescribeRouteTables(&ec2.DescribeRouteTablesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("association.subnet-id"),
+				Values: []*string{createdInstance.SubnetId},
+			},
+		},
+	})
+	if err != nil || len(routeTables.RouteTables) < 1 {
+		assert.Fail(t, fmt.Sprintf("error finding route table for subnet %s, %v", *createdInstance.SubnetId, err))
+		return
+	}
+
+	for _, route := range routeTables.RouteTables[0].Routes {
+		igws, err := awsProvider.EC2.DescribeInternetGateways(&ec2.DescribeInternetGatewaysInput{
+			InternetGatewayIds: []*string{route.GatewayId},
+		})
+		if err == nil && len(igws.InternetGateways) > 0 {
+			return
+		}
+	}
+	assert.Fail(t, "subnet associated is not a public subnet")
 }
 
-// checkIfInfraTagExists checks if the infrastructure tag exists on the created instance
-func checkIfInfraTagExists(instance *ec2.Instance, infraID string) bool {
-	if instance == nil {
-		return false
+// checkInstanceIsAttachedWithName asserts if an instance has a Name tag value.
+// Instance input needs to be updated before use.
+func checkInstanceIsAttachedWithName(t *testing.T) {
+	for _, tag := range createdInstance.Tags {
+		if *tag.Key == "Name" && tag.Value != nil {
+			return
+		}
 	}
+	assert.Fail(t, "instance is not assigned a name")
+}
+
+// checkInstanceInfraTagExists asserts if the infrastructure tag exists on the created instance.
+// Instance input needs to be updated before use.
+func checkInstanceInfraTagExists(t *testing.T) {
 	key := "kubernetes.io/cluster/" + infraID
 	value := "owned"
-	for _, tag := range instance.Tags {
+	for _, tag := range createdInstance.Tags {
 		if *tag.Key == key && *tag.Value == value {
-			return true
+			return
+		}
+	}
+	assert.Fail(t, "infrastructure tag not found")
+}
+
+// checkInstanceHasClusterWorkerSg asserts if the created instance has OpenShift cluster worker security group attached.
+func checkInstanceHasClusterWorkerSg(t *testing.T) {
```

Comment:
Usually all the methods start with test..

Reply:
I'll change them, I kinda changed them from test to check.